lib/dark_worlds_server/matchmaking/matching_coordinator.ex

```diff
@@ -35,17 +35,25 @@ defmodule DarkWorldsServer.Matchmaking.MatchingCoordinator do
     session_ref = make_ref()
     Process.send_after(self(), {:check_timeout, session_ref}, @start_game_timeout_ms)
     players = [{user_id, from}]
+    notify_players_amount(players, @session_player_amount)
+
     {:reply, :ok, %{state | players: players, session: session_ref}}
   end
 
   def handle_call({:join, user_id}, {from, _}, state) do
-    players = state.players ++ [{user_id, from}]
-    send(self(), :check_capacity)
-    {:reply, :ok, %{state | players: players}}
+    if Enum.any?(state.players, fn {player_user_id, _} -> user_id == player_user_id end) do
+      {:reply, :ok, state}
+    else
+      players = state.players ++ [{user_id, from}]
+      notify_players_amount(players, @session_player_amount)
```

Comment:
If you do this here it can happen that there are more than `@session_player_amount` in `players` at that time

Remember that this is a list we keep filling and on `:check_capacity` we take the first N of those, so no guarantees on size

Reply:
Should be fixed now!